In [17]:
import pandas as pd
df = pd.read_csv('final_result.csv', low_memory = False)
df.head()

id        title  score_val category            descript  \
0    35849217  쭈꾸미블루스 신촌본점       4.59    주꾸미요리  신촌 맛집 날치알쌈과 매운 쭈꾸미   
1  1749849331   금고기비스트로 신촌        NaN  육류,고기요리                 NaN   
2  1198813653     댄싱홍콩 신촌점        NaN      중식당                 NaN   
3  1071728472      정통집 신촌점       4.32   돼지고기구이                 NaN   
4  1491059869     오적회관 신촌점        NaN       한식                 NaN   

                                             reviews  keyword1   keyword2  \
0  노포식당요즘 인기있는곳이래서 왔어요 주꾸미 삼겹 꽃게새우사리추가해서.먹었는데요 양념...  음식이 맛있어요       친절해요   
1  지나가는데 너무 예뻐서 술집인 줄 알았는데 알고 보니까 신촌 삼겹살 맛집이더라구요 ...  음식이 맛있어요  고기 질이 좋아요   
2  협력사들과 회의 끝내고 오찬 같이했네요  간만에 오니 새로운 메뉴도 생기고.. 사람...  음식이 맛있어요  인테리어가 멋져요   
3  이 리뷰는 업체로부터 무료 이용권을 제공 받은 대가로 작성한 솔직 후기입니다.  h...  음식이 맛있어요     양이 많아요   
4  📺'생방송투데이'와 '맛의 승부사'에도 방영될 정도로 검증된 유명한 💢핫플매장이더라...  음식이 맛있어요     양이 많아요   

      keyword3  
0    인테리어가 멋져요  
1         친절해요  
2  특별한 메뉴가 있어요  
3         친절해요  
4     재료가 신선해요

In [18]:
print(df.columns)

Index(['id', 'title', 'score_val', 'category', 'descript', 'reviews',
       'keyword1', 'keyword2', 'keyword3'],
      dtype='object')


In [19]:
sum(df.duplicated())

0

In [20]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 694 entries, 0 to 693
Data columns (total 9 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   id         694 non-null    int64  
 1   title      694 non-null    object 
 2   score_val  411 non-null    float64
 3   category   694 non-null    object 
 4   descript   197 non-null    object 
 5   reviews    688 non-null    object 
 6   keyword1   670 non-null    object 
 7   keyword2   670 non-null    object 
 8   keyword3   670 non-null    object 
dtypes: float64(1), int64(1), object(7)
memory usage: 48.9+ KB


In [21]:
df.isna().sum()

id             0
title          0
score_val    283
category       0
descript     497
reviews        6
keyword1      24
keyword2      24
keyword3      24
dtype: int64

##❗전체 데이터 개수: 694
### 수작업으로 보강해야 할 데이터: keyword(670/694), reviews(688/694)
### 그 외 보강해야 할 데이터: score_val(411/694), descript(197/694)
#### 📌 descript -> 제외 고려


In [22]:
# 키워드 누락된 행 찾기
df[df.keyword1.isna( )]

id                title  score_val category descript  \
502  1953092293                  소수헌        NaN      일식당      NaN   
554  1199259251                커스텀아토        NaN    케이크전문      NaN   
572  1527827739              국진남 이대점        NaN       한식      NaN   
573  1596231490  벤앤제리스 아이스크림 신촌이대DV점        NaN   카페,디저트      NaN   
579  1489998688         도네누볏짚등삼겹살이대점        NaN   돼지고기구이      NaN   
583  1876526492         원조얼큰샤브샤브 이대점        NaN       한식      NaN   
592  1422957079         병천미도순대국 이대본점        NaN   순대,순댓국      NaN   
593  1875409867   카페아이엔지 이화여대 국제교육관점        NaN   카페,디저트      NaN   
602  1441087834       양반댁 묵은지김치찜 이대점        NaN       한식      NaN   
616    38276479             테이스트 이대점        NaN   카페,디저트      NaN   
628  1427980003           뉴스보이커피 이대점        NaN   카페,디저트      NaN   
636  1910994068               이대꼬치꼬치       4.35    오뎅,꼬치      NaN   
639   206256821         신세계푸드FS이화여대점       4.25     푸드코트      NaN   
653  1580038253           고래맥주창고 이대점        NaN    맥주,호프      NaN   
654  1976483551            스윗드로우 케이크        NaN    케이크전문      NaN   
658  1604677319               단풍뼈칼국수        NaN   칼국수,만두      NaN   
659  1533544820              파티션 WSC       4.69   카페,디저트      NaN   
672  1183833041        냠냠족발&보쌈 마포아현점        NaN    족발,보쌈      NaN   
674  1116546793                 바니카페        NaN       카페      NaN   
675  1971177172             에보니앤아이보리        NaN   카페,디저트      NaN   
678  1278760975                 독도물회        NaN       한식      NaN   
682  1517647458                문득커피바        NaN       카페      NaN   
683  1343790146                   캐피        NaN   카페,디저트      NaN   
693    89888882               보라네도시락        NaN   도시락,컵밥      NaN   

                                               reviews keyword1 keyword2  \
502  비 오는 날에 첫 방문한 소수헌 런치. 비오는 것도 운치로 느껴질 정도로 멋진 한옥...      NaN      NaN   
554  좋아요 / 가격도 합리적이고 너무 맛잇어요 남편이 좋아했어요 / 두번째 주문하는데,...      NaN      NaN   
572  냉면 그릇 생각보다 큽니다.!! 그리고 진짜 맛있어요 ㅠㅠ / 송정 3대 국밥 - ...      NaN      NaN   
573  긴가민가했는데 2층 계단 올라가니까 있네요. 방문 현장주문 가능해요. 양 꽉꽉 눌러...      NaN      NaN   
579                                                NaN      NaN      NaN   
583                                                NaN      NaN      NaN   
592  맛있게 먹을 점심 메뉴를 고르다가 익숙한 음식이어서 실패할 일은 없겠다는 마음으로 ...      NaN      NaN   
593  이대 대학교회 옆 건물인 국제관에 있는 카페인데 편안하고 밝은 에너지가 느껴지는 분...      NaN      NaN   
602                      ㅎㅎㅎㅎㅎㅎㅎㅎㅎㅎ / 매콤한 김치찜이 땡길때 찾는집      NaN      NaN   
616                                                NaN      NaN      NaN   
628               주변에 일이있어서 잠깐차한잔하려고 들른곳 편안하고 커피도 맛있당~      NaN      NaN   
636  so so / 좋아요 / 좋아요 / 😀 / 판매할때는 양이 많아 보였는데 빼서 주신...      NaN      NaN   
639  굿 / 좋아요 / 굿 / 소떡소떡이 매우 맛있음 / 생활관 학식 콩나물제육비빔밥 맛...      NaN      NaN   
653  세 번째 구매입니다 ㅋㅋㅋㅋ 싸게 독특한 맥주를 살 수 있어서 좋아용! / 크러쉬도...      NaN      NaN   
654  달디달고 달디달고 달디단 케이크 🎂  적정한 단맛! 초안이 없어서 고민했는데 원하는...      NaN      NaN   
658  밥이 무한리필이라서 좋음. 칼국수가 싫으신 분이라면 주문시 국수는 빼달라고 말씀드리...      NaN      NaN   
659  조아요 / 괜찮아요. / 커피가 참 맛있었고, 편안하면서도 정돈된 이 공간만의 특유...      NaN      NaN   
672        쫄깃쫄깃 맛있었어요 막국수도 보쌈도 쭈꾸미도 족박돌 맛있었어요 번창하세요🩷🩷🩷      NaN      NaN   
674               맛이따 / 굿 / 친절해요 / 넘귀여운 눈사람이네요~~~감사합니다      NaN      NaN   
675                                                NaN      NaN      NaN   
678                                  너모 맛잇숨미다 ㅠ 또먹을예정😏      NaN      NaN   
682                                                NaN      NaN      NaN   
683  이대역 바로 근처 조용한 골목에 위치해 있는 신상카페 캐피  덴마크 코펜하겐 스페셜...      NaN      NaN   
693                                                NaN      NaN      NaN   

    keyword3  
502      NaN  
554      NaN  
572      NaN  
573      NaN  
579      NaN  
583      NaN  
592      NaN  
593      NaN  
602      NaN  
616      NaN  
628      NaN  
636      NaN  
639      NaN  
653      NaN  
654      NaN  
658      NaN  
659      NaN  
672      NaN  
674      NaN  
675      NaN  
678      NaN  
682      NaN  
683      NaN  
693      NaN

In [23]:
# 키워드 없는 행 삭제
df2 = df.dropna(subset = ['keyword1', 'keyword2', 'keyword3'])

In [24]:
df2[df2.keyword1.isna( )]

Empty DataFrame
Columns: [id, title, score_val, category, descript, reviews, keyword1, keyword2, keyword3]
Index: []

In [25]:
# 리뷰 누락된 행 찾기
df2[df2.reviews.isna( )]

Empty DataFrame
Columns: [id, title, score_val, category, descript, reviews, keyword1, keyword2, keyword3]
Index: []

### 💥 키워드, 리뷰 없는 행 삭제 완료

In [26]:
df2

id         title  score_val  category            descript  \
0      35849217   쭈꾸미블루스 신촌본점       4.59     주꾸미요리  신촌 맛집 날치알쌈과 매운 쭈꾸미   
1    1749849331    금고기비스트로 신촌        NaN   육류,고기요리                 NaN   
2    1198813653      댄싱홍콩 신촌점        NaN       중식당                 NaN   
3    1071728472       정통집 신촌점       4.32    돼지고기구이                 NaN   
4    1491059869      오적회관 신촌점        NaN        한식                 NaN   
..          ...           ...        ...       ...                 ...   
688  1774558523          베지피티        NaN  다이어트,샐러드                 NaN   
689  1056075705   1987신샤브 마포점        NaN      샤브샤브                 NaN   
690    38705128  미스터 참치 마포직영점        NaN       생선회                 NaN   
691  1934848485          Quid        NaN        와인                 NaN   
692  1283043137           즐겁다        NaN     맥주,호프                 NaN   

                                               reviews   keyword1   keyword2  \
0    노포식당요즘 인기있는곳이래서 왔어요 주꾸미 삼겹 꽃게새우사리추가해서.먹었는데요 양념...   음식이 맛있어요       친절해요   
1    지나가는데 너무 예뻐서 술집인 줄 알았는데 알고 보니까 신촌 삼겹살 맛집이더라구요 ...   음식이 맛있어요  고기 질이 좋아요   
2    협력사들과 회의 끝내고 오찬 같이했네요  간만에 오니 새로운 메뉴도 생기고.. 사람...   음식이 맛있어요  인테리어가 멋져요   
3    이 리뷰는 업체로부터 무료 이용권을 제공 받은 대가로 작성한 솔직 후기입니다.  h...   음식이 맛있어요     양이 많아요   
4    📺'생방송투데이'와 '맛의 승부사'에도 방영될 정도로 검증된 유명한 💢핫플매장이더라...   음식이 맛있어요     양이 많아요   
..                                                 ...        ...        ...   
688  이대역 혼밥 맛집 발견~~~!!! 신촌박스퀘어 계속 가봐야지 가봐야지 하다가 드디어...   음식이 맛있어요   재료가 신선해요   
689  아파트주민들이 아이들과 많이오는 느낌ㅋㅋㅋㅋ아기의자하나있고 식기도 요청하면 주십니다...   음식이 맛있어요   재료가 신선해요   
690  생일 저녁으로 남자친구랑 동네 가까운 곳에서 뭐먹을지 고민하다가 갑자기 참치가 땡겨...   음식이 맛있어요   재료가 신선해요   
691  친구와 너무 편안하고 즐거운 시간 보냈습니다. 음식도 너무 다 맛있고 와인도 잘어울...  인테리어가 멋져요       친절해요   
692  바로 집앞이라 자주 가는 술집이에요. 납작만두 야채 완전 새콤하네요 ㅎㅎ 만두가 1...   음식이 맛있어요       친절해요   

        keyword3  
0      인테리어가 멋져요  
1           친절해요  
2    특별한 메뉴가 있어요  
3           친절해요  
4       재료가 신선해요  
..           ...  
688         친절해요  
689         친절해요  
690     가성비가 좋아요  
691     대화하기 좋아요  
692  특별한 메뉴가 있어요  

[670 rows x 9 columns]

In [27]:
# descript 열 drop
df3 = df2.drop(['descript'], axis = 1)

In [28]:
df3

id         title  score_val  category  \
0      35849217   쭈꾸미블루스 신촌본점       4.59     주꾸미요리   
1    1749849331    금고기비스트로 신촌        NaN   육류,고기요리   
2    1198813653      댄싱홍콩 신촌점        NaN       중식당   
3    1071728472       정통집 신촌점       4.32    돼지고기구이   
4    1491059869      오적회관 신촌점        NaN        한식   
..          ...           ...        ...       ...   
688  1774558523          베지피티        NaN  다이어트,샐러드   
689  1056075705   1987신샤브 마포점        NaN      샤브샤브   
690    38705128  미스터 참치 마포직영점        NaN       생선회   
691  1934848485          Quid        NaN        와인   
692  1283043137           즐겁다        NaN     맥주,호프   

                                               reviews   keyword1   keyword2  \
0    노포식당요즘 인기있는곳이래서 왔어요 주꾸미 삼겹 꽃게새우사리추가해서.먹었는데요 양념...   음식이 맛있어요       친절해요   
1    지나가는데 너무 예뻐서 술집인 줄 알았는데 알고 보니까 신촌 삼겹살 맛집이더라구요 ...   음식이 맛있어요  고기 질이 좋아요   
2    협력사들과 회의 끝내고 오찬 같이했네요  간만에 오니 새로운 메뉴도 생기고.. 사람...   음식이 맛있어요  인테리어가 멋져요   
3    이 리뷰는 업체로부터 무료 이용권을 제공 받은 대가로 작성한 솔직 후기입니다.  h...   음식이 맛있어요     양이 많아요   
4    📺'생방송투데이'와 '맛의 승부사'에도 방영될 정도로 검증된 유명한 💢핫플매장이더라...   음식이 맛있어요     양이 많아요   
..                                                 ...        ...        ...   
688  이대역 혼밥 맛집 발견~~~!!! 신촌박스퀘어 계속 가봐야지 가봐야지 하다가 드디어...   음식이 맛있어요   재료가 신선해요   
689  아파트주민들이 아이들과 많이오는 느낌ㅋㅋㅋㅋ아기의자하나있고 식기도 요청하면 주십니다...   음식이 맛있어요   재료가 신선해요   
690  생일 저녁으로 남자친구랑 동네 가까운 곳에서 뭐먹을지 고민하다가 갑자기 참치가 땡겨...   음식이 맛있어요   재료가 신선해요   
691  친구와 너무 편안하고 즐거운 시간 보냈습니다. 음식도 너무 다 맛있고 와인도 잘어울...  인테리어가 멋져요       친절해요   
692  바로 집앞이라 자주 가는 술집이에요. 납작만두 야채 완전 새콤하네요 ㅎㅎ 만두가 1...   음식이 맛있어요       친절해요   

        keyword3  
0      인테리어가 멋져요  
1           친절해요  
2    특별한 메뉴가 있어요  
3           친절해요  
4       재료가 신선해요  
..           ...  
688         친절해요  
689         친절해요  
690     가성비가 좋아요  
691     대화하기 좋아요  
692  특별한 메뉴가 있어요  

[670 rows x 8 columns]

### 💥 descript 열 삭제 완료

In [29]:
df3.info()

<class 'pandas.core.frame.DataFrame'>
Index: 670 entries, 0 to 692
Data columns (total 8 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   id         670 non-null    int64  
 1   title      670 non-null    object 
 2   score_val  408 non-null    float64
 3   category   670 non-null    object 
 4   reviews    670 non-null    object 
 5   keyword1   670 non-null    object 
 6   keyword2   670 non-null    object 
 7   keyword3   670 non-null    object 
dtypes: float64(1), int64(1), object(6)
memory usage: 47.1+ KB


In [36]:
df3.isna().sum()

id             0
title          0
score_val    262
category       0
reviews        0
keyword1       0
keyword2       0
keyword3       0
dtype: int64

In [39]:
# 별점 결측치 평균으로 대체
df3.loc[df3['score_val'] != df3['score_val'], 'score_val'] = df3['score_val'].mean()

In [40]:
df3

id         title  score_val  category  \
0      35849217   쭈꾸미블루스 신촌본점   4.590000     주꾸미요리   
1    1749849331    금고기비스트로 신촌   4.494412   육류,고기요리   
2    1198813653      댄싱홍콩 신촌점   4.494412       중식당   
3    1071728472       정통집 신촌점   4.320000    돼지고기구이   
4    1491059869      오적회관 신촌점   4.494412        한식   
..          ...           ...        ...       ...   
688  1774558523          베지피티   4.494412  다이어트,샐러드   
689  1056075705   1987신샤브 마포점   4.494412      샤브샤브   
690    38705128  미스터 참치 마포직영점   4.494412       생선회   
691  1934848485          Quid   4.494412        와인   
692  1283043137           즐겁다   4.494412     맥주,호프   

                                               reviews   keyword1   keyword2  \
0    노포식당요즘 인기있는곳이래서 왔어요 주꾸미 삼겹 꽃게새우사리추가해서.먹었는데요 양념...   음식이 맛있어요       친절해요   
1    지나가는데 너무 예뻐서 술집인 줄 알았는데 알고 보니까 신촌 삼겹살 맛집이더라구요 ...   음식이 맛있어요  고기 질이 좋아요   
2    협력사들과 회의 끝내고 오찬 같이했네요  간만에 오니 새로운 메뉴도 생기고.. 사람...   음식이 맛있어요  인테리어가 멋져요   
3    이 리뷰는 업체로부터 무료 이용권을 제공 받은 대가로 작성한 솔직 후기입니다.  h...   음식이 맛있어요     양이 많아요   
4    📺'생방송투데이'와 '맛의 승부사'에도 방영될 정도로 검증된 유명한 💢핫플매장이더라...   음식이 맛있어요     양이 많아요   
..                                                 ...        ...        ...   
688  이대역 혼밥 맛집 발견~~~!!! 신촌박스퀘어 계속 가봐야지 가봐야지 하다가 드디어...   음식이 맛있어요   재료가 신선해요   
689  아파트주민들이 아이들과 많이오는 느낌ㅋㅋㅋㅋ아기의자하나있고 식기도 요청하면 주십니다...   음식이 맛있어요   재료가 신선해요   
690  생일 저녁으로 남자친구랑 동네 가까운 곳에서 뭐먹을지 고민하다가 갑자기 참치가 땡겨...   음식이 맛있어요   재료가 신선해요   
691  친구와 너무 편안하고 즐거운 시간 보냈습니다. 음식도 너무 다 맛있고 와인도 잘어울...  인테리어가 멋져요       친절해요   
692  바로 집앞이라 자주 가는 술집이에요. 납작만두 야채 완전 새콤하네요 ㅎㅎ 만두가 1...   음식이 맛있어요       친절해요   

        keyword3  
0      인테리어가 멋져요  
1           친절해요  
2    특별한 메뉴가 있어요  
3           친절해요  
4       재료가 신선해요  
..           ...  
688         친절해요  
689         친절해요  
690     가성비가 좋아요  
691     대화하기 좋아요  
692  특별한 메뉴가 있어요  

[670 rows x 8 columns]

In [41]:
df3.isna().sum()

id           0
title        0
score_val    0
category     0
reviews      0
keyword1     0
keyword2     0
keyword3     0
dtype: int64

### 💥 리뷰 결측치 채우기 완료

In [44]:
df4 = df3.round(2)

In [45]:
df4

id         title  score_val  category  \
0      35849217   쭈꾸미블루스 신촌본점       4.59     주꾸미요리   
1    1749849331    금고기비스트로 신촌       4.49   육류,고기요리   
2    1198813653      댄싱홍콩 신촌점       4.49       중식당   
3    1071728472       정통집 신촌점       4.32    돼지고기구이   
4    1491059869      오적회관 신촌점       4.49        한식   
..          ...           ...        ...       ...   
688  1774558523          베지피티       4.49  다이어트,샐러드   
689  1056075705   1987신샤브 마포점       4.49      샤브샤브   
690    38705128  미스터 참치 마포직영점       4.49       생선회   
691  1934848485          Quid       4.49        와인   
692  1283043137           즐겁다       4.49     맥주,호프   

                                               reviews   keyword1   keyword2  \
0    노포식당요즘 인기있는곳이래서 왔어요 주꾸미 삼겹 꽃게새우사리추가해서.먹었는데요 양념...   음식이 맛있어요       친절해요   
1    지나가는데 너무 예뻐서 술집인 줄 알았는데 알고 보니까 신촌 삼겹살 맛집이더라구요 ...   음식이 맛있어요  고기 질이 좋아요   
2    협력사들과 회의 끝내고 오찬 같이했네요  간만에 오니 새로운 메뉴도 생기고.. 사람...   음식이 맛있어요  인테리어가 멋져요   
3    이 리뷰는 업체로부터 무료 이용권을 제공 받은 대가로 작성한 솔직 후기입니다.  h...   음식이 맛있어요     양이 많아요   
4    📺'생방송투데이'와 '맛의 승부사'에도 방영될 정도로 검증된 유명한 💢핫플매장이더라...   음식이 맛있어요     양이 많아요   
..                                                 ...        ...        ...   
688  이대역 혼밥 맛집 발견~~~!!! 신촌박스퀘어 계속 가봐야지 가봐야지 하다가 드디어...   음식이 맛있어요   재료가 신선해요   
689  아파트주민들이 아이들과 많이오는 느낌ㅋㅋㅋㅋ아기의자하나있고 식기도 요청하면 주십니다...   음식이 맛있어요   재료가 신선해요   
690  생일 저녁으로 남자친구랑 동네 가까운 곳에서 뭐먹을지 고민하다가 갑자기 참치가 땡겨...   음식이 맛있어요   재료가 신선해요   
691  친구와 너무 편안하고 즐거운 시간 보냈습니다. 음식도 너무 다 맛있고 와인도 잘어울...  인테리어가 멋져요       친절해요   
692  바로 집앞이라 자주 가는 술집이에요. 납작만두 야채 완전 새콤하네요 ㅎㅎ 만두가 1...   음식이 맛있어요       친절해요   

        keyword3  
0      인테리어가 멋져요  
1           친절해요  
2    특별한 메뉴가 있어요  
3           친절해요  
4       재료가 신선해요  
..           ...  
688         친절해요  
689         친절해요  
690     가성비가 좋아요  
691     대화하기 좋아요  
692  특별한 메뉴가 있어요  

[670 rows x 8 columns]

In [47]:
df4.isna().sum()

id           0
title        0
score_val    0
category     0
reviews      0
keyword1     0
keyword2     0
keyword3     0
dtype: int64

In [46]:
df4.to_csv("final_preprocessing.csv", index=False)

### 📌 실행 후 최종파일 다운로드